docs/getting_started.rst

```diff
@@ -0,0 +1,296 @@
+
+Genetic methods
+~~~~~~~~~~~~~~~
+
+Genetic methods in sgkit are nearly always applied to individual ``Dataset`` objects.  For a full list of
+available methods, see :ref:`api_methods`.
+
+In this example, the ``variant_stats`` method is applied to a dataset to compute a number of statistics
+across samples for each individual variant:
+
+.. ipython:: python
+
+    sg.variant_stats(ds, merge=False)
+
+There are two ways that the results of every function are handled -- either they are merged with the provided
+dataset or they are returned in a separate dataset.  See :ref:`dataset_merge` for more details.
+
+
+.. _missing_data:
+
+Missing data
+~~~~~~~~~~~~
+
+Missing data in sgkit is represented using a sentinel value within data arrays
+(``-1`` in integer arrays and ``NaN`` in float arrays) as well as a companion boolean mask array
+(``True`` where data is missing). The sentinel values are often more useful when implementing compiled
+operations while the boolean mask array facilitates user operations in a higher level API like Xarray or Numpy.
+
+This example shows how either can be used, though users should prefer the mask array where possible since
+its on-disk representation is typically far smaller after compression is applied.
+
+.. ipython:: python
+
+    dsm = simulate_genotype_call_dataset(n_variant=1, n_sample=4, n_ploidy=2, missing_pct=.3, seed=4)
+    dsm.call_genotype
+
+    # Count alternate alleles while omitting partial calls
+    ##############
+    # Using Xarray
+    ##############
+    import xarray as xr
+    alt_allele_count = xr.where(
+        # Identify where there are any missing calls across chromosomes
+        dsm.call_genotype_mask.any(dim='ploidy'),
+        -1, # Return -1 if any one call for a chromosome is missing
+        (dsm.call_genotype > 0).sum(dim='ploidy') # Otherwise, sum non-ref calls
+    )
+    # Note that only the first two samples have meaningful counts since
+    # at least one call is missing for the last two samples
+    alt_allele_count.values
+
+    #############
+    # Using Numba
+    #############
+    import numba
+    import numpy as np
+
+    def alt_allele_count(gt):
+        out = np.full(gt.shape[:2], -1, dtype=np.int64)
+        for i, j in np.ndindex(*out.shape):
+            if np.all(gt[i, j] >= 0):
+                out[i, j] = np.sum(gt[i, j] > 0)
+        return out
+
+    # Jit-compiled functions are often simpler with a single array input, since
+    # conditional logic based on sentinel values is easier to program with this API
+    numba.njit(alt_allele_count)(dsm.call_genotype.values)
+
+This is not necessarily a level of detail most users should need to worry about. Missing data
+is handled explicitly in sgkit functions and where this isn't possible, limitations related
+to it are documented along with potential workarounds.
+
+Chaining operations
+~~~~~~~~~~~~~~~~~~~
+
+This example shows to chain multiple sgkit, xarray, and pandas operations into a single pipeline:
```

Comment:
Is there some upstream documentation/tutorial explaining what an xarray/pandas pipeline is? I think this would be helpful here.

Reply:
https://github.com/pystatgen/sgkit/commit/ae349eea8e8b2094f7d55bf282e5c15d74c4a84b#diff-9bc8bf6e8d9db020cefce1fbc0426795R265